### DISCUSSION TITLE
Custom Provider: How can I make my deployments use vars defined in the "custom_provider"?

### DISCUSSION THREAD

**Discussion by cgiraldo** [D10S0VSkY-OSS/Stack-Lifecycle-Deployment#159] [Q&A]
Hi,
First of all, nice work!

I am trying to use SLD with vsphere as a provider. I would like to configure some common vars in the custom_provider form (e.g., vsphere endpoint, username and password, datastore, folder...).

```
{
"TF_VAR_vsphere_endpoint": "192.168.20.19",
"TF_VAR_vsphere_username": "Administrator@vsphere.local",
"TF_VAR_vsphere_password": "***********",
"TF_VAR_vsphere_insecure_connection": "true",
"TF_VAR_vsphere_datacenter": "MCCE_CPD",
"TF_VAR_vsphere_cluster": "MCCE_cluster",
"TF_VAR_vsphere_datastore": "Datastore2",
"TF_VAR_vsphere_network": "mcce",
"TF_VAR_vsphere_folder": "mcce-devel"
}
```
However, when I deploy my stacks, SLD creates the form from my variables.tf including these variables, and if I leave them empty in the form, it set the variable to the empty string (I would like to use the value set in "TF_VAR_xxxx" in the custom_provider).

am I missing something? 
How can I make my deployments use vars defined in the "custom_provider"?

Thanks!

**Comment by D10S0VSkY-OSS** [comment]
Hi @cgiraldo,
SLD reads the file variables.tf, in this file you build the form that you want to present to the user in UI.
For this case I recommend that you have a file variables_hidden.tf
Regards!

**Comment by cgiraldo** [comment]
That will work for us.

Thanks!
